### DISCUSSION TITLE
Recommended way to type useQuery data if we know it cannot be undefined?

### DISCUSSION THREAD

**Discussion by taylorkline** [reduxjs/redux-toolkit#1493] [Q&A] [answered]
I have a query that is ran in an `AppLoader` component to fetch some app-wide account data. Unless that data is successfully retrieved, none of its children could possibly be rendered.

Given that, how can I intelligently type `data` from `useGetAccountQuery` / `useGetAccountQueryState` to indicate that it cannot possibly be `undefined`?

Right now I have:

```typescript
  // @ts-ignore: data will be AccountResponse here
  const { data }: { data: AccountResponse } = useGetAccountQuery();
```

But I know that this is completely silly and unsafe since, for example, changing the `ResultType` of the `getAccount` endpoint would not raise any warnings.

How can I just type `data` as `ResultType` instead of `ResultType | undefined`?

This is probably trivial to anyone with more familiarity with TypeScipt than I, so thank you in advance for your help.

**Comment by phryneas** [comment, marked as answer]
Hmm. Something like
```ts
function assertHasData<T>(x: {data?: T}): asserts x is {data: T} {}


const result = useGetAccountQuery();
assertHasData(result)

// result.data is defined here
```
or
```ts
function assertIsDefined<T>(x: T|undefined): asserts x is T {}


const { data } = useGetAccountQuery();
assertIsDefined(data)

// data is defined here
```

**Reply by taylorkline** [reply]
Thanks, super helpful!
